engine/src/mblandroiddc.cpp

```diff
@@ -2012,6 +2018,29 @@ JNIEXPORT void JNICALL Java_com_runrev_android_Engine_doCreate(JNIEnv *env, jobj
 	s_android_activity = env -> NewGlobalRef(activity);
 	MCLog("Got global android activity: %p\n", s_android_activity);
 	
+    // Cache the class loader
+    jclass t_object_class = env->FindClass("java/lang/Object");
+    
+    jmethodID t_get_class = env->GetMethodID(t_object_class, "getClass",
+                                             "()Ljava/lang/Class;");
+    
+    jobject t_activity_class =
+        env->CallObjectMethod(s_android_activity, t_get_class);
+    
+    jclass t_class_class = env->FindClass("java/lang/Class");
+    
+    jmethodID t_get_class_loader =
+        env->GetMethodID(t_class_class, "getClassLoader",
+                                  "()Ljava/lang/ClassLoader;");
+    
+    jobject t_class_loader =
+        env->CallObjectMethod(t_activity_class, t_get_class_loader);
+    
+    MCLog("Got class loader: %p\n", t_class_loader);
+    
+    // The class loader - make sure we hold a global ref.
+    s_android_class_loader = env->NewGlobalRef(t_class_loader);
```

Comment:
Do we need to delete the global ref on shutdown? (if we are doing so for other such things, that is).